Fix Probed Framerate for videos with non constant framerate.

06needhamt opened the pull request:
Fixes an issue where slow motion videos taken on certain android devices do not have a constant framerate have their framerate reported incorrectly by ffprobe. We can fix this issue by calculating the framerate ourselves by using the nb_frames and duration values returned by ffprobe. The video file included in this PR is an example of a video which suffers from this issue. Converting this video using the latest Xabe build causes FFmpeg to duplicate frames endlessly since this library reports the framerate to be 90000 FPS when the real frame rate is 116.234 FPS.

I have updated the basic conversion methods to force set the framerate to stop this issue from occurring although i don't know if there is anywhere else i need to update.

06needhamt commented:
@tomaszzmuda Please could I have your thoughts / review on this potentially breaking change?

06needhamt commented:
@tomaszzmuda Thanks, which tests are failing? They all pass on my local machine.

tomaszzmuda commented:
![image](https://user-images.githubusercontent.com/13612984/85407967-317f3a00-b564-11ea-96e0-64ce01befb4e.png)

tomaszzmuda commented:
More details:
![image](https://user-images.githubusercontent.com/13612984/85411157-48279000-b568-11ea-913a-94e9d2414b1f.png)
![image](https://user-images.githubusercontent.com/13612984/85411170-4b228080-b568-11ea-966b-5ae5904c2e8c.png)

06needhamt commented:
@tomaszzmuda I believe I have fixed the tests. On the SloMo one there was an issue when you force the output format to be h264 FFmpeg produces a corrupt file. I don't know if this is something we need to investigate or whether i should report it to FFmpeg. and for the Gif test the framerate was getting tested against 25 and it now returns 24.889 in the new method so I have updated the test accordingly.

06needhamt commented:
@tomaszzmuda Looks like a test failed again, but i'm not sure which one. Is there anyway you can make the test results public?

tomaszzmuda commented:
![image](https://user-images.githubusercontent.com/13612984/85509044-8ae37980-b5f5-11ea-8276-3124e9a7aabe.png)
I've changed project type to public, you should see build results now

06needhamt commented:
@tomaszzmuda Not sure what's going on with the `ToGifTest`, where it says it expects 24.889, so I change the value to 24.889 then it reported it expects 25 so I change it back and then it expects 24.889 again. Please could you take a look? Thanks

tomaszzmuda commented:
ToGitTest has 2 test datasets. 
![image](https://user-images.githubusercontent.com/13612984/85812223-5cce7880-b760-11ea-9327-5e459067f23e.png)
First run fails because second set has different framerate than first one.
Skipping first frame results different framerate so I have just added framerate as a parameter there.
